src/main/scala/loamstream/drm/Drmaa1Client.scala

```diff
@@ -113,16 +116,18 @@ final class Drmaa1Client(nativeSpecBuilder: NativeSpecBuilder) extends DrmaaClie
    * @return Success wrapping the JobStatus corresponding to the code obtained from DRM,
    * or Failure if the job id isn't known.  (Lamely, this can occur if the job is finished.)
    */
-  override def statusOf(jobId: String): Try[DrmStatus] = {
+  override def statusOf(taskId: DrmTaskId): Try[DrmStatus] = {
     Try {
       withSession { session =>
-        val status = session.getJobProgramStatus(jobId)
+        //TODO Is .jobId right here?  Do we need to care about the index as well?
+        val status = session.getJobProgramStatus(taskId.jobId)
```

Comment:
I will admit some confusion here... above when doing the ordering for jobs, the code first compared `jobId` and - if equal - then went to `taskId`. That made me think that a task was a a piece of a job. But, this code makes it appear as though a job is part of a task?

I have a meeting now, but if these can get answered by you, it would likely help me in reviewing the rest of the PR when I continue it :smile:.

Reply:
It's surprising that this works, I'll investigate.  This whole rigamarole is due to changes on the Uger side: we used to be able specify job ids as strings encoded like "${job id of the entire task array}.${index within task array}", which is what Uger gives us, and accepts.  What's new is that it doesn't accept that format always; sometimes the index needs to be tracked and passed separately.  It's been enough weeks since I wrote this that I need to look back on what happened.

While this stuff is confusing and ought to be straightened out, please prioritize `ExecutionState` and the related changes to `RxExecuter` if there's not time for everything.  `ExecutionState` represents a bunch of complex new work, with thread-safety implications, and could _really_ use another pair of eyes.